Gems/Atom/Tools/MaterialEditor/Code/Source/Document/MaterialDocument.cpp

```diff
@@ -551,25 +551,40 @@ namespace MaterialEditor
         // Assets must still be used for now because they contain the final accumulated value after all other materials
         // in the hierarchy are applied
         bool enumerateResult = m_materialTypeSourceData.EnumeratePropertyGroups(
-            [this, &parentPropertyValues](
-                const AZStd::string& propertyIdContext, const AZ::RPI::MaterialTypeSourceData::PropertyGroup* propertyGroup)
+            [this, &parentPropertyValues](const AZ::RPI::MaterialTypeSourceData::PropertyGroupStack& propertyGroupStack)
             {
-                // Add any material functors that are located inside each property group.
-                if (!AddEditorMaterialFunctors(propertyGroup->GetFunctors()))
+                using namespace AZ::RPI;
+
+                const MaterialTypeSourceData::PropertyGroup* propertyGroup = propertyGroupStack.back();
+                
+                MaterialNameContext groupNameContext = MaterialTypeSourceData::MakeMaterialNameContext(propertyGroupStack);
+
+                if (!AddEditorMaterialFunctors(propertyGroup->GetFunctors(), groupNameContext))
                 {
                     return false;
                 }
 
+                AZStd::vector<AZStd::string> groupNameVector;
+                AZStd::vector<AZStd::string> groupDisplayNameVector;
+
+                for (auto& group : propertyGroupStack)
+                {
+                    groupNameVector.push_back(group->GetName());
+                    groupDisplayNameVector.push_back(group->GetDisplayName());
+                }
```

Comment:
That's kind of a long story.

First, groupDisplayNameVector cannot reasonably be done with MaterialNameContext. At first I tried to add support for display names to MaterialNameContext but it started to break down. MaterialNameContext is used both at tool time and at runtime. It is serialized with LuaMaterialFunctor (and potentially other functors in the future) so that names can be contextualized as the scripts run. If we add support for display names to LuaMaterialFunctor then we either have to skip this field in the Reflect() function, or be careful to exclude this data when saving the MaterialNameContext in the LuaMaterialFunctor. Another option could be to have two separate MaterialNameContext classes, with one dedicated to tools with the extra data. I couldn't come up with something I liked so I just gave up and decided to keep the oddity isolated to these couple lines of code.

groupNameVector _sort of_ could be replaced with a MaterialNameContext, but we can't use the groupNameContext created above because that includes the whole group stack. We would need to make another MaterialNameContext for everything _except_ the last group, because MaterialNameContext::ContextualizeProperty tacks on the last group. Since I had to make this loop anyway for the sake of groupDisplayNameVector, I decided it was simplest to just go ahead and handle groupNameVector in the same way.